### ISSUE TITLE
Sync with latest telegram source code

### ISSUE THREAD

**Issue by Graveyard9988** [Dr4iv3rNope/NotSoAndroidAyuGram#4]
I'd be grateful if you sync your fork with latest Telegram source code please

**Comment by Dr4iv3rNope** [comment]
Repository will not be synced with latest Telegram source code unless [original AyuGram project](https://github.com/AyuGram/AyuGram4A) will sync it.

I think AyuGram will not update their **open repository** soon, because [exteraGram](https://github.com/exteraSquad/exteraGram) and [AyuGram](https://github.com/AyuGram/AyuGram4A) are **currently developed in private repositories**. AlexeyZavar (the main contributor of AyuGram) said in AyuGram chat that **he will update open repository when every feature will be "stable enough"**. I may be wrong but it seems that AlexeyZavar is not interested in open repository anymore.

_P. S. from [the AyuGram FAQ](https://t.me/ayugramchat/3557/48244): "We publish sources for [AyuGram on GitHub](https://github.com/AyuGram). But as for Android, I delay pushing commits until the main [exteraGram repo](https://github.com/exteraSquad/exteraGram) gets updated."_

**Comment by ZavaruKitsu** [comment]
@Dr4iv3rNope nah, you're too bad at reading (or interpreting, idk) FAQs
https://t.me/ayugramchat/3557/48244
Also, I told you in chat everything about open source, but you're still acting like "ohh, he's an evil!!!"
Don't be like that, man.

**Comment by Dr4iv3rNope** [comment]
@ZavaruKitsu, I don't want to argue with you and continue dispute.

Regarding _"ohh, he's an evil!!!"_ comment:

1. Everything started with jokes about "backdoor in AyuGram" in the AyuGram chat, after you have added AyuSync.
2. The backend of AyuSync is closed source because of "security issues". I don't trust this feature, and i don't want to share my "HWID", user ids and etc.
3. Because of the "lack of proprietary source files" the original AyuGram project cannot be built, therefore AyuSync cannot be removed.
4. APKs that are being uploaded to the Telegram channel cannot be trusted, because the source code doesn't match the public repository.

**Comment by ZavaruKitsu** [comment]
Nah, you mustn't be reading the chat, I wrote that ayusync will be open sourced once finished. Because of this, I hide it on both desktop and android.
And "proprietary" folder will be open sourced too, just like I wrote in the chat.

And, well, not going to argue about "cannot be trusted", think really whatever you want. I'm done arguing with open source cultists.

**Comment by Graveyard9988** [comment]
I wish you could sync your fork with newest codes of official telegram
Please revise your desicion 

Thanks for your explaining

**Comment by Graveyard9988** [comment]
Could you sync with newest Extergram or Ayugram please?

**Comment by Dr4iv3rNope** [comment]
> Could you sync with newest Extergram or Ayugram please?

There are no new commits in these projects. Nothing to sync.

**Comment by Graveyard9988** [comment]
Oh they have released apk but haven't updated source code

**Comment by rushizgithub** [comment]
> Oh they have released apk but haven't updated source code

those are beta releases , which are based on private src

**Comment by mightatnight** [comment]
> Nah, you mustn't be reading the chat, I wrote that ayusync will be open sourced once finished. Because of this, I hide it on both desktop and android.
> And "proprietary" folder will be open sourced too, just like I wrote in the chat.
> 
> And, well, not going to argue about "cannot be trusted", think really whatever you want. I'm done arguing with open source cultists.

How's it going, are you still finishing up?))

**Comment by Qing0731** [comment]
@ZavaruKitsu It has been a year since you said those words, and there has been no submission to the Ayugram repository since then. Your so-called open-source Ayusync project has also been silent, and in fact, you have merely found an excuse to deceive users by not releasing the source code. The AyuHistoryHook.java file, which you deliberately withheld to prevent others from removing the tracker, is another example of your hypocrisy.

**Comment by mightatnight** [comment]
> @ZavaruKitsu It has been a year since you said those words, and there has been no submission to the Ayugram repository since then. Your so-called open-source Ayusync project has also been silent, and in fact, you have merely found an excuse to deceive users by not releasing the source code. The AyuHistoryHook.java file, which you deliberately withheld to prevent others from removing the tracker, is another example of your dishonesty.


This a Russian developer,nothing new :)

**Comment by Keeperorowner** [comment]
Thanks to your work, there is now a fully open-source client called NagramX with the TOS Broken feature, and there's no need to use the closed-source Ayugram anymore.😊
https://github.com/risin42/NagramX

**Comment by Artisan-memory** [comment]
Can I trust NagramX? Is there smth like AyuSync or else?

**Comment by Keeperorowner** [comment]
You can fully trust NagramX. It is completely open source and can be built locally, and there is nothing like ayusync that leaks privacy
